examples/platforms/posix/radio.c

```diff
@@ -308,16 +309,24 @@ void otPlatRadioSetPromiscuous(bool aEnable)
 void posixRadioInit(void)
 {
     struct sockaddr_in sockaddr;
+    char *offset;
     memset(&sockaddr, 0, sizeof(sockaddr));
     sockaddr.sin_family = AF_INET;
 
+    offset = getenv("PORT_OFFSET");
+
+    if (offset)
+    {
+        sPortOffset = strtol(offset, NULL, 0) * WELLKNOWN_NODE_ID;
```

Comment:
You could use the `endptr` arg.  From the man page:

     If endptr is not NULL, strtol() stores the address of the first invalid
     character in *endptr.  If there were no digits at all, however, strtol()
     stores the original value of str in *endptr.  (Thus, if *str is not `\0'
     but **endptr is `\0' on return, the entire string was valid.)